src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,457 @@
+        else {
+            final DetailAST grandParent = parentAst.getParent();
+            final DetailAST scopeAst = grandParent.findFirstToken(TokenTypes.SLIST);
+            if (scopeAst == null) {
+                result = grandParent;
+            }
+            else {
+                result = scopeAst;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is
+     * used as left-hand side value. An identifier is being used as a left-hand side
+     * value if it is used as the left operand of an assignment or as an
+     * operand of a stand-alone increment or decrement.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return True if identAst is used as a left-hand side value otherwise false.
+     */
+    private static boolean isLeftHandSideValue(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return isStandAloneIncrementOrDecrement(identAst)
+                || parent.getType() == TokenTypes.ASSIGN
+                && identAst != parent.getLastChild();
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is used as
+     * an operand of a stand-alone increment or decrement.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return True if identAst is used as an operand of stand-alone
+     *         increment or decrement otherwise false
+     */
+    private static boolean isStandAloneIncrementOrDecrement(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        final DetailAST grandParent = parent.getParent();
+        return TokenUtil.isOfType(parent, INCREMENT_AND_DECREMENT_TOKENS)
+                && TokenUtil.isOfType(grandParent, TokenTypes.EXPR)
+                && (!TokenUtil.isOfType(grandParent.getParent(),
+                TokenTypes.ELIST, TokenTypes.INDEX_OP, TokenTypes.ASSIGN)
+                || grandParent.getParent().getParent().getType() == TokenTypes.FOR_ITERATOR);
+    }
+
+    /**
+     * Maintains information about the local variable.
+     */
+    private static final class VariableDesc {
+
+        /**
+         * The name of the variable.
+         */
+        private final String name;
+
+        /**
+         * DetailAst of type {@link TokenTypes#TYPE}.
+         */
+        private final DetailAST typeAst;
+
+        /**
+         * The scope of variable is determined by the DetailAst of type
+         * {@link TokenTypes#SLIST} which is enclosing the variable.
+         */
+        private final DetailAST scope;
+
+        /**
+         * Is the variable used again.
+         */
+        private boolean used;
+
+        /**
+         * Create a new VariableDesc instance.
+         *
+         * @param name name of the variable
+         * @param typeAst DetailAst of type {@link TokenTypes#TYPE}
+         * @param scope {@link VariableDesc#scope}
```

Comment:
It would be better to not have to look at private class members to determine what a `scope` is in this context.

Reply:
removed it.

Reply:
Changed it.

Reply:
Resolved.